rpm/0003-Fix-Calendar-updateNotebook-event-visibility-updates.patch

```diff
@@ -0,0 +1,29 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Pekka Vuorela <pekka.vuorela@jolla.com>
+Date: Thu, 21 Mar 2024 11:38:56 +0200
+Subject: [PATCH] Fix Calendar::updateNotebook event visibility updates
+
+Broken by commit 6a18355546. Earlier this was using
+QMultiHash::values(key) for updates but the commit made it loop over
+all the events, thus when some notebook was updating visibility,
+everything got updated accordingly.
+---
+ src/calendar.cpp | 5 ++++-
+ 1 file changed, 4 insertions(+), 1 deletion(-)
+
+diff --git a/src/calendar.cpp b/src/calendar.cpp
+index 44146500e..516e9f8e8 100644
+--- a/src/calendar.cpp
++++ b/src/calendar.cpp
+@@ -339,7 +339,10 @@ bool Calendar::updateNotebook(const QString &notebook, bool isVisible)
+         return false;
+     } else {
+         d->mNotebooks.insert(notebook, isVisible);
+-        for (auto noteIt = d->mNotebookIncidences.cbegin(); noteIt != d->mNotebookIncidences.cend(); ++noteIt) {
++
++        for (auto  noteIt = d->mNotebookIncidences.find(notebook);
++             noteIt != d->mNotebookIncidences.end() && noteIt.key() == notebook;
```

Comment:
I'm not getting something.

If there is max one item to tweak -> for should not be needed.
If there can be multiple items to tweak -> I'd expect "if(noteIt.key() != notebook) continue;" within the loop.

Or is there somekind of quarantee that ++ iteration happens in order, so that matching keys are seen one after another without non-matching items in between?

Reply:
Apparently there is, or at least this matches what QMultiHash documentation gives as the way to do it.